scripts/gen-sample-config.js

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2019 New Vector Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// The sample config is generated as follows:
+// 1. Start with SdkConfig's DEFAULTS
+// 2. Overwrite with features and Settings defaults
+// 3. Overwrite with samples defined here
+
+const SdkConfig = require("matrix-react-sdk/lib/SdkConfig");
+const fs = require("fs");
+
+const SAMPLES = {
+    default_server_name: "matrix.org",
+    disable_custom_urls: false,
+    disable_guests: false,
+    disable_login_language_selector: false,
+    disable_3pid_login: false,
+    brand: "Riot",
+    welcomeUserId: "@riot-bot:matrix.org",
+    default_federate: true,
+    bug_report_endpoint_url: "https://riot.im/bugreports/submit",
+    integrations_ui_url: "https://scalar.vector.im/",
+    integrations_rest_url: "https://scalar.vector.im/api",
+    integrations_jitsi_widget_url: "https://scalar.vector.im/api/widgets/jitsi.html",
+    integrations_widgets_urls: [
+        "https://scalar-staging.riot.im/scalar/api",
+        "https://scalar.vector.im/api",
+    ],
+    default_theme: "light",
+    roomDirectory: {
+        servers: [
+            "matrix.org",
+        ],
+    },
+    enable_presence_by_hs_url: {
+        "https://matrix.org": false,
+    },
+    piwik: {
+        siteId: 1,
+        url: "https://piwik.riot.im",
+        whitelistedHSUrls: ["https://matrix.org"],
+        whitelistedISUrls: ["https://vector.im", "https://matrix.org"],
+    },
+    embeddedPages: {
+        homeUrl: "home.html",
+        welcomeUrl: "welcome.html",
+    },
+    cross_origin_renderer_url: "https://usercontent.riot.im/v1.html",
+    branding: {
+        authHeaderLogoUrl: "themes/riot/img/logos/riot-im-logo-black-text.svg",
+        welcomeBackgroundUrl: "themes/riot/img/backgrounds/valley.jpg",
+    },
+    update_base_url: "https://riot.im/download/desktop/update/",
+    sync_timeline_limit: 8,
+    terms_and_conditions_links: [
+        {
+            url: "https://matrix.org/docs/guides/code_of_conduct",
+            text: "matrix.org code of conduct",
+        },
+    ],
+};
+
+function parseSettings() {
+    // This is by far the worst and cleanest way to load the settings config.
```

Comment:
Hmm, agreed this is unfortunate... Is there any way to avoid the `eval`? Some possibilities:

* Move `SETTINGS` object out of `Settings.js` without the controllers and such (not great, since likely means bits of settings in multiple places)
* This script could set an environments variable, and then `Settings.js` could check this at import time and replace dependencies with no-op stubs to allow a normal `require` to complete